### DISCUSSION TITLE
custom hook is taking too long to load - timeout

### DISCUSSION THREAD

**Discussion by daksamit** [strapi/strapi#8067] [Legacy Topics]
Hi!
Beta version of strapi allowed us to set timeout property for hook config in hook.json.
Now we have hook.js and timeout doesn't work. It's intended or bug?
For example I have sometimes this error on strapi load:
`error (hook: mycustomhook) is taking too long to load.` 

Config file `config/hook.js` like that doesn't work?
```js
module.exports = ({ env }) => ({
  settings: {
    mycustomhook: {
      enabled: true,
      timeout: 6000,
      options: {}
    },
  },
});

**Comment by MattieBelt** [comment]
Hey @daksamit,

Per hook `timeout` config isn't supported (also wasn't in beta). You can only config a `timeout` for all your hooks, for example:
```js
module.exports = ({ env }) => ({
  timeout: 2000                     // default 1000
  settings: {
    mycustomhook: {
      enabled: true,
      options: {}
    },
  },
});
```

**Reply by MattieBelt** [reply]
I liked the idea of a per hook timeout config, I just submitted a PR for it, check it out #8081

**Reply by daksamit** [reply]
Very nice, thanks!
